hydromt/models/model.py

```diff
@@ -313,127 +300,58 @@ def update(
-        opt = opt or {}
-        opt = self._check_get_opt(opt)
+        steps = steps or {}
+        validate_steps(self, steps)
+
+        # check if region.create is in the steps, and remove it.
+        if steps.pop("region.create", None) is not None:
+            self.logger.warning(
+                "region.create can only be called when building a model."
+            )
 
         # read current model
         if not self.root.is_writing_mode():
             if model_out is None:
                 raise ValueError(
-                    '"model_out" directory required when updating in "read-only" mode'
+                    '"model_out" directory required when updating in "read-only" mode.'
                 )
             self.read()
-            if forceful_overwrite:
-                self.root.set(model_out, mode="w+")
-            else:
-                self.root.set(model_out, mode="w")
+            mode = "w+" if forceful_overwrite else "w"
+            self.root.set(model_out, mode=mode)
 
         # check if model has a region
         if self.region.data is None:
             raise ValueError("Model region not found, setup model using `build` first.")
 
-        # remove CLI_ARGS method from options and throw warning
-        method = self._CLI_ARGS["region"]
-        if method in opt:
-            opt.pop(method)  # remove from opt
-            self.logger.warning(f'"{method}" can only be called when building a model.')
-
-        # loop over other methods from config file
-        for method in opt:
-            # if any write_* functions are present in opt, skip the final self.write()
-            if "write" in method:
-                write = False
-            kwargs = {} if opt[method] is None else opt[method]
-            self._run_log_method(method, **kwargs)
-
-        # write
-        if write:
+        # loop over methods from config file
+        for step in steps:
+            self.logger.info(f"update: {step}")
+            kwargs = steps[step] or {}
+            # Call the methods.
+            rgetattr(self, step)(kwargs)
+
+        # If there are any write options included in the steps,
+        # we don't need to write the whole model.
+        if write and not self._options_contain_write(steps):
             self.write()
 
         self._cleanup(forceful_overwrite=forceful_overwrite)
 
-    ## file system
-    def _assert_write_mode(self):
-        if not self.root.is_writing_mode():
-            raise IOError("Model opened in read-only mode")
-
-    def _assert_read_mode(self):
-        if not self.root.is_reading_mode():
-            raise IOError("Model opened in write-only mode")
-
-    # I/O
-    def read(
-        self,
-        components: List = None,
-    ) -> None:
-        """Read the complete model schematization and configuration from model files.
+    @hydromt_step
+    def write(self, components: Optional[list[str]] = None):
+        """Write all components of the model to disk with defaults."""
```

Comment:
Could you detail a bit more with Parameters etc?
For example by default all components are written but if components is not None, then only the listed selection will be written